### PR TITLE
use highway network and employ decaying learning rate

### PR DESCRIPTION
**Pull request by primetang**
use highway network and employ decaying learning rate in the new code.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -126,15 +134,21 @@
         checkpoint_prefix = os.path.join(checkpoint_dir, "model")
         if not os.path.exists(checkpoint_dir):
             os.makedirs(checkpoint_dir)
-        saver = tf.train.Saver(tf.global_variables())
+        try:
+            saver = tf.train.Saver(tf.global_variables())
```
Comment (dennybritz) [inline train.py]: Why is this needed? Can't you just say `saver = tf.train.Saver()`?

**Comment by dennybritz** [comment]
Also, I think `batches_num_pre_epoch` is somewhat of a strange name. I think it should be `num_batches_per_epoch`. I also think it would be cleaner to not decay per epoch, but decay per some number of steps and have a parameter for that and get rid of that batches per epoch function.

**Comment by primetang** [comment]
Thank you for your attention, I have rewrite the code about `decay_step`, and use `saver = tf.train.Saver()` instead of `try/catch`.

**Comment by qwaider** [comment]
Interested
